x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -182,6 +182,9 @@ export const signalRulesAlertType = ({
       try {
         const inputIndex = await getInputIndex(services, version, index);
         const privileges = await checkPrivileges(services, inputIndex);
+        const timestampFields: string[] = params.timestampOverride
```

Comment:
This file must be getting difficult for everyone to read as people are missing things like the params are destructoring above this which was kind of us following the advice of the Kibana guidelines to destructor things rather than use their params.xyz to cut down on noise.

So right now we have two different ways of accessing state which is `params.timestampOverride` and then in the other spots we directly use the destructored form.

You can use the destructored one if you want like so:

```ts
        const timestampFields: string[] =
          timestampOverride != null ? ['@timestamp', timestampOverride] : ['@timestamp'];
```

I use a `!= null` there because the way you have it, it will evaluate an empty string as being false. Maybe that's what you want though?

For example:
```ts
❯ node
> '' ? true : false
false
> 'hi' ? true : false
true
> 
```

If you want explicit checks against empty string I would suggest maybe `!isEmpty()`?

But regardless I leave it up to you on the `params.xyz` vs the destructored form above but really we should cut down on duplicate state within functions to avoid confusion if we can.

Reply:
Good catch I have always tried to use != null since it is more obvious what we are checking for. This one slipped through though. Thanks. And yeah I should be using the destructured `timestampOverride`. This file absolutely could use some TLC though..